docs/CLIUsage.md

```diff
@@ -1,23 +1,120 @@
+| `kubectl kudo plan status [flags]` | View all available plans. |
+| `kubectl kudo plan history <name> [flags]` | View all available plans. |
+
+### Flags
+
+|  Name | Shorthand | Default | Usage  | 
+|---|---|---|---|
+| repo-version | | | A specific repo version on the official GitHub repo; defaults to the most recent |
+| all-dependencies |  | true | Installs all dependency frameworks (with frameworkversions) it can find from the dependency section in `frameworkversions`|
+| auto-approve | | false | Skip interactive approval when existing version found. To always install the latest framework/frameworkversion |  
+| githubcredential | | | The file path to github credential file; defaults to $HOME/.git-credentials |
+| instance | | Framework name | The instance name. (default to Framework name) |
+| namespace | | "default" | The namespace where the operator watches for changes |
+
+### Environment Variables
+
+Environment Variables override flags and are intended to give the user more customizable CLI options.
+
+|  Environment Variable | Description  | 
+|---|---|
+| `GIT_USER`  |  Set a GitHub user to connect via the GitHub API with |
+| `GIT_PASSWORD` | Set a GitHub password to connect via the GitHub API with | 
+
+### Examples
+
+#### Install a Framework
+
+There are two options installing a Framework. While `kubectl apply -f *.yaml` is encouraged when developing and
+having `framework.yaml` or `frameworkversion.yaml` locally present, it is recommended to use the official Frameworks 
+provided in our [kubebuilder/frameworks](https://github.com/kudobuilder/frameworks) repo. `Kudoctl` offers a convenient
+way of installing those files in the command line.
+
+##### Install just the Framework without dependencies
 
-- Kudoctl
-- Kubectl Kudo Plugin
+This is the default behavior and just installs the according Framework.
 
-## Kudoctl
+```bash
+$ kubectl kudo install kafka
+framework.kudo.k8s.io/v1alpha1/kafka created
+frameworkversion.kudo.k8s.io/v1alpha1/kafka-2.11-2.4.0 created
+```
+
+#### Install a Framework with Dependencies
 
-The usage is intended to make your life easier when working with `Kudo`. A work flow would look like this:
+Sometimes you want to automatically install all dependencies a Framework comes with. This behavior is controlled via
+the flag `--all-dependencies`, and if set installs all dependencies a Framework has as well.
 
-1) You get a list of all available Instances deployed by `Kudo`
-2) You get the status the particular Instance of interest
-3) You see a history to a specific Instance
+```bash
+$ kubectl kudo install kafka --all-dependencies
+framework.kudo.k8s.io/v1alpha1/kafka created
+frameworkversion.kudo.k8s.io/v1alpha1/kafka-2.11-2.4.0 created
+framework.kudo.k8s.io/v1alpha1/zookeeper created
+frameworkversion.kudo.k8s.io/v1alpha1/zookeeper-1.0 created
+```
+
+#### Install a Framework with InstanceName & Parameters
+
+Use `--instance` and `--paramter`/`-p` for setting an Instance name or Parameter:
```

Comment:
I don't think so. It changes the name of the particular `Instance` CRD but when using `--name` it could also implicate that it changes the names for the entire stack for it (`Framework`, `FrameworkVersion` and `Instance`). Ergo I was hoping that with `--instance` it is clear that we mean just the `Instance` CRD name.